### PR TITLE
Move telemetry to inline code for use in templates

### PR DESCRIPTION
**Pull request by chcosta**
We want to support this model for teams participating in Arcade templates - https://github.com/dotnet/arcade-minimalci-sample#base-your-builds-on-arcade-for-ease-of-use

This will unblock enabling telemetry for the [CLICommandLineParser repo](https://github.com/dotnet/CliCommandLineParser/blob/master/eng/build.yml#L9).  (FYI @livarcocc)

Repo reuse templates don't download extra scripts, so any scripting must be inline.

I'm not aware of any consumers actively using the non-inlined scripts at this time, so removing them.

Validated telemetry being sent with [devdiv.vs.com build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=1958029&_a=summary&view=logs) and checking SQL database entries.

@alexperovich @jonfortescue

### CODE REVIEW

File: eng/common/templates/steps/telemetry-start.yml [hunk 1]
```diff
@@ -11,8 +11,52 @@ steps:
...
+      echo "Failed To Send Job Start information"
+      # We have to append the ## vso prefix or vso will pick up the command when it dumps the inline script into the shell
+      vstsLogOutput="vso[task.logissue type=error;sourcepath=telemetry/start-job.sh;code=1;]Failed to Send Job Start information: $curlStatus"
+      echo "##$vstsLogOutput"
+      exit 1
+    fi
+    
+    # Set the Helix_JobToken variable
+    export Helix_JobToken=`echo $curlResult | xargs echo` # Strip Quotes
+    echo "##vso[task.setvariable variable=Helix_JobToken;issecret=true;]$Helix_JobToken"
```
Comment (alexperovich) [inline eng/common/templates/steps/telemetry-start.yml]: I think you need to do the same thing here that you did on line 51

File: eng/common/templates/steps/telemetry-start.yml [hunk 2]
```diff
@@ -35,11 +97,22 @@ steps:
...
     }
-  workingDirectory: $(Build.SourcesDirectory)/eng/common
+    
+    $jobInfoJson = $jobInfo | ConvertTo-Json
+
+    Write-Host "Job Info: $jobInfoJson"
+    $jobToken = Invoke-RestMethod -Uri "https://helix.dot.net/api/2018-03-14/telemetry/job?access_token=$($env:HelixApiAccessToken)" -Method Post -ContentType "application/json" -Body $jobInfoJson
+        
+    $env:Helix_JobToken = $jobToken
+    Write-Host "##vso[task.setvariable variable=Helix_JobToken;issecret=true;]$env:Helix_JobToken"
```
Comment (alexperovich) [inline eng/common/templates/steps/telemetry-start.yml]: Same thing here

File: eng/common/templates/steps/telemetry-start.yml [hunk 3]
```diff
@@ -52,11 +125,11 @@ steps:
...
-    ./telemetry/build/start.ps1 -BuildUri $env:BuildUri
-    if ($LASTEXITCODE -ne 0) {
-      exit $LASTEXITCODE
-    }
-  workingDirectory: $(Build.SourcesDirectory)/eng/common
+    $workItemId = Invoke-RestMethod -Uri "https://helix.dot.net/api/2018-03-14/telemetry/job/build?buildUri=$([Net.WebUtility]::UrlEncode($env:BuildUri))" -Method Post -ContentType "application/json" -Body "" `
+      -Headers @{ 'X-Helix-Job-Token'=$env:Helix_JobToken }
+  
+    $env:Helix_WorkItemId = $workItemId
+    Write-Host "##vso[task.setvariable variable=Helix_WorkItemId]$env:Helix_WorkItemId"
```
Comment (alexperovich) [inline eng/common/templates/steps/telemetry-start.yml]: here too

File: eng/common/templates/steps/telemetry-start.yml (hunk 1 above)
Comment (chcosta) [reply eng/common/templates/steps/telemetry-start.yml]: This ends up not really doing anything valid if it gets printed.  The other line actually reports to VSTS that there's an error.  I debated doing that but didn't think it was worth it.  I'm ok with just following the pattern for consistency though.

Comment (alexperovich) [reply eng/common/templates/steps/telemetry-start.yml]: It probably sets the variable which then gets reset when the command actually runs. I'm fine with leaving it like this then.

**Pull request merged**